kafka-3.8.yaml

```diff
@@ -74,6 +74,31 @@ subpackages:
           ln -s /usr/lib/kafka/config ${{targets.subpkgdir}}/opt/bitnami/kafka/config
           ln -s /usr/lib/kafka/logs ${{targets.subpkgdir}}/opt/bitnami/kafka/logs
 
+  - name: kafka-bitnami-kraft-compat-${{vars.major-minor-version}}
+    description: "compat package with bitnami/kafka image compatible with KRaft mode"
+    dependencies:
+      provides:
+        - kafka-bitnami-kraft-compat=${{package.full-version}}
+      runtime:
+        - wait-for-port
+        - bash # some helper scripts use bash
+        - openjdk-17-default-jvm
+    pipeline:
+      - uses: bitnami/compat
+        with:
+          image: kafka
+          version-path: 3.8/debian-12
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}
+          mkdir ${{targets.subpkgdir}}/docker-entrypoint-initdb.d/
+          mkdir -p ${{targets.subpkgdir}}/opt/bitnami/kafka/
+          mkdir -p ${{targets.subpkgdir}}/usr/lib/kafka/logs
+          cp -r ${{targets.destdir}}/usr/lib/kafka/bin ${{targets.subpkgdir}}/opt/bitnami/kafka
+          cp -r ${{targets.destdir}}/usr/lib/kafka/libs ${{targets.subpkgdir}}/opt/bitnami/kafka
+          cp -r ${{targets.destdir}}/usr/lib/kafka/logs ${{targets.subpkgdir}}/opt/bitnami/kafka
+          cp -r ${{targets.destdir}}/usr/lib/kafka/config ${{targets.subpkgdir}}/opt/bitnami/kafka
+          mv ${{targets.subpkgdir}}/opt/bitnami/kafka/config/server.properties ${{targets.subpkgdir}}/opt/bitnami/kafka/config/server.properties.original
```

Comment:
in bitnami image i think all modes are co-located, do we really need a separate build of compat package; then then also a separate build of image?

As in, can this all not be folded into kafka-bitnami-compat-${{vars.major-minor-version}} which we now know is incomplete?

Reply:
Thanks for the review. I think this PR is superseded via https://github.com/wolfi-dev/os/pull/29341 (thanks for @srishtih) 

It was an ownership issue so far we tracked it. 

We are renaming the file `server.properties` to `server.properties.original` in the main package and when we are running it then `server.properties` file is getting created at runtime with `kafka:kafka` ownership while the application runs.

After renaming things, docker-compose setup, package level and image level tests are passing so we can drop this extra `compat` package which we were building for kafka kraft mode.

Reply:
i like that other PR better it looks smaller! but it is marked as draft. Reviewed that other one, and will park this one for now.